erigon-lib/downloader/mdbx_piece_completion.go

```diff
@@ -33,17 +35,31 @@ const (
 )
 
 type mdbxPieceCompletion struct {
-	db kv.RwDB
+	db        *mdbx.MdbxKV
+	mu        sync.RWMutex
+	completed map[infohash.T]*roaring.Bitmap
 }
 
 var _ storage.PieceCompletion = (*mdbxPieceCompletion)(nil)
 
 func NewMdbxPieceCompletion(db kv.RwDB) (ret storage.PieceCompletion, err error) {
-	ret = &mdbxPieceCompletion{db: db}
+	ret = &mdbxPieceCompletion{db: db.(*mdbx.MdbxKV), completed: map[infohash.T]*roaring.Bitmap{}}
 	return
 }
 
-func (m mdbxPieceCompletion) Get(pk metainfo.PieceKey) (cn storage.Completion, err error) {
+func (m *mdbxPieceCompletion) Get(pk metainfo.PieceKey) (cn storage.Completion, err error) {
```

Comment:
as i know we don't use `mdbxPieceCompletion`. we using `NewMdbxPieceCompletionBatch`.